packages/fulfillment/src/services/fulfillment-module-service.ts

```diff
@@ -1557,4 +1566,66 @@ export default class FulfillmentModuleService<
 
     return geoZoneConstraints
   }
+
+  protected aggregateFulfillmentSetCreatedEvents(
+    createdFulfillmentSets: TEntity[],
+    sharedContext: Context
+  ): void {
+    const buildMessage = ({
+      eventName,
+      id,
+      object,
+    }: {
+      eventName: string
+      id: string
+      object: string
+    }) => {
+      return {
+        eventName,
+        metadata: {
+          object,
+          service: Modules.FULFILLMENT,
+          action: "created",
+          eventGroupId: sharedContext.eventGroupId,
+        },
+        data: { id },
+      }
+    }
+
+    for (const fulfillmentSet of createdFulfillmentSets) {
+      sharedContext.messageAggregator!.saveRawMessageData(
+        buildMessage({
+          eventName: FulfillmentUtils.FulfillmentEvents.created,
+          id: fulfillmentSet.id,
+          object: "fulfillment_set",
+        })
+      )
+
+      if (fulfillmentSet.service_zones?.length) {
```

Comment:
We can have an early return here, and below as well

Reply:
simplified a bit more

Reply:
wdyt?